sdk/lib/rtl/rangelist.c

```diff
@@ -13,546 +13,1556 @@
- * 	RtlCopyRangeList
+    RtlEntry = RtlpEntryFromLink(RangeList->ListHead.Flink);
+    NextRtlEntry = RtlpEntryFromLink(RtlEntry->ListEntry.Flink);
+
+    while (&RangeList->ListHead != &RtlEntry->ListEntry && End >= RtlEntry->Start)
+    {
+        DPRINT("RtlDeleteRange: %p [%I64X-%I64X] %p\n", RtlEntry, RtlEntry->Start, RtlEntry->End, Owner);
+
+        if (RtlEntry->PrivateFlags & RTLP_ENTRY_IS_MERGED)
+        {
+            if (Start >= RtlEntry->Start && End <= RtlEntry->End)
+            {
+                MergedRtlEntry = RtlpEntryFromLink(RtlEntry->Merged.ListHead.Flink);
+                NextMergedRtlEntry = RtlpEntryFromLink(MergedRtlEntry->ListEntry.Flink);
+
+                while (&RtlEntry->Merged.ListHead != &MergedRtlEntry->ListEntry)
+                {
+                    if (MergedRtlEntry->Start == Start &&
+                        MergedRtlEntry->End == End &&
+                        MergedRtlEntry->Allocated.Owner == Owner)
+                    {
+                        Status = RtlpDeleteFromMergedRange(MergedRtlEntry, RtlEntry);
+                        goto Exit;
+                    }
+
+                    MergedRtlEntry = NextMergedRtlEntry;
+                    NextMergedRtlEntry = RtlpEntryFromLink(MergedRtlEntry->ListEntry.Flink);
+                }
+            }
+        }
+        else if (RtlEntry->Start == Start &&
+                 RtlEntry->End == End &&
+                 RtlEntry->Allocated.Owner == Owner)
+        {
+            RemoveEntryList(&RtlEntry->ListEntry);
+
+            DPRINT("RtlDeleteRange: Free %p [%I64X-%I64X] %p\n", RtlEntry, RtlEntry->Start, RtlEntry->End, Owner);
+            //ExFreeToPagedLookasideList(&RtlpRangeListEntryLookasideList, RtlEntry);
+            RtlpFreeMemory(RtlEntry, 'elRR');
+
+            Status = STATUS_SUCCESS;
+            break;
+        }
+
+        RtlEntry = NextRtlEntry;
+        NextRtlEntry = RtlpEntryFromLink(RtlEntry->ListEntry.Flink);
+    }
+
+Exit:
+
+    if (NT_SUCCESS(Status))
+    {
+        RangeList->Count++;
+        RangeList->Stamp++;
+    }
+
+    DPRINT("RtlDeleteRange: return Status %X\n", Status);
+    return Status;
+}
+
+/**
+ * @name RtlGetNextRange
  *
- * DESCRIPTION
- *	Copy a range list.
+ * Retrieves the next (or previous) range of a range list.
  *
- * ARGUMENTS
- *	CopyRangeList	Pointer to the destination range list.
- *	RangeList	Pointer to the source range list.
+ * @param Iterator
+ * Pointer to a user supplied list state buffer.
  *
- * RETURN VALUE
- *	Status
+ * @param OutRange
+ * Address of pointer to the first range.
  *
- * @implemented
+ * @param MoveForwards
+ * TRUE - get next range, FALSE - get previous range.
+ *
+ * @return
+ * Returns NTSTATUS.
  */
+CODE_SEG("PAGE")
+NTSYSAPI
 NTSTATUS
 NTAPI
-RtlCopyRangeList(OUT PRTL_RANGE_LIST CopyRangeList,
-                 IN PRTL_RANGE_LIST RangeList)
+RtlGetNextRange(
+    _Inout_ PRTL_RANGE_LIST_ITERATOR Iterator,
+    _Out_ PRTL_RANGE * OutRange,
```

Comment:
```suggestion
    _Outptr_ PRTL_RANGE *Range,
```
Also, for consistency with the other RtlGet***Range functions (+ what's already in the RTL header), I'm suggesting renaming the `OutRange` parameter to just `Range`.